src/Language/PureScript/Ide/Rebuild.hs

```diff
@@ -37,23 +35,25 @@ rebuildFile path = do
          Right [m] -> pure m
          Right _ -> throwError . GeneralError $ "Please define exactly one module."
 
+  -- Externs files must be sorted ahead of time, so that they get applied
+  -- correctly to the 'Environment'.
   externs <- sortExterns m . M.delete (P.getModuleName m) =<< getExternFiles
 
   outputDirectory <- confOutputPath . envConfiguration <$> ask
 
-  let foreignModule = replaceExtension path "js"
-  foreignExists <- liftIO (doesFileExist foreignModule)
+  -- Infer foreign modules using their file paths
+  let filePathMap = M.singleton (P.getModuleName m) (Left P.RebuildAlways)
+  foreigns <- P.inferForeignModules filePathMap
 
-  let ma = P.buildMakeActions outputDirectory
-                              (M.singleton (P.getModuleName m) (Left P.RebuildAlways))
-                              (if foreignExists
-                                 then M.singleton (P.getModuleName m) foreignModule
-                                 else M.empty)
-                              False
+  -- Silence progress update messages during the build
+  let actions = (P.buildMakeActions outputDirectory filePathMap foreigns False)
```

Comment:
These changes will conflict with my cache-rebuild branch since I already made similar changes to the action building